src/jdk.compiler/share/classes/com/sun/tools/javac/comp/TransPatterns.java

```diff
@@ -344,12 +345,11 @@ public void visitRecordPattern(JCRecordPattern tree) {
         result = test != null ? test : makeLit(syms.booleanType, 1);
     }
 
-    private MethodSymbol getAccessor(RecordComponent component) {
+    private MethodSymbol getAccessor(DiagnosticPosition pos, RecordComponent component) {
         return component2Proxy.computeIfAbsent(component, c -> {
-            MethodSymbol realAccessor = (MethodSymbol) component.owner
-                                                 .members()
-                                                 .findFirst(component.name, s -> s.kind == Kind.MTH &&
-                                                                                 ((MethodSymbol) s).params.isEmpty());
+            MethodSymbol realAccessor =
+                    rs.resolveInternalMethod(pos, env, component.owner.type,
```

Comment:
why not just reading the accessor field in the record component? it should point to the method symbol of the accessor

Reply:
Thanks, I didn't realize that! Fixed now.